addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/EchoHandler.java

```diff
@@ -145,6 +155,15 @@ public boolean setDeviceAndUpdateThingState(AccountHandler accountHandler, @Null
             return false;
         }
         this.device = device;
+        String[] capabilities = device.capabilities;
+        if (capabilities != null) {
+            this.capabilities.clear();
+            for (String capability : capabilities) {
+                if (capability != null) {
+                    this.capabilities.add(capability);
+                }
+            }
+        }
```

Comment:
How about a one-liner using the Java 8 functional API? :)
Like (out of my mind): 
`this.capabilities = Stream.of(device.capabilities).filter(c -> c != null).collect(Collectors.toSet())`

Reply:
If you are suggesting to use streams do it the right way :smile:
`this.capabilities = Stream.of(device.capabilities).filter(Objects::nonNull).collect(Collectors.toSet());`

Reply:
Thank you for this hint, I searched for this pattern already, because I know this from .NET LINQ, but I did not find it in Java.

Reply:
Done at this line. But I will do this in the future for all of this pattern. This makes it much more easier. Thank you!